modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5186,6 +5186,23 @@ bool GDScriptAnalyzer::is_type_compatible(const GDScriptParser::DataType &p_targ
 	return false;
 }
 
+bool GDScriptAnalyzer::is_super_type(const GDScriptParser::DataType &p_left_type, const GDScriptParser::DataType &p_right_type) {
+	if (p_left_type.is_variant()) {
+		return false;
+	}
+	if (p_left_type.is_variant() || !p_right_type.is_hard_type()) {
+		return true;
+	}
+	if (!is_type_compatible(p_left_type, p_right_type, true)) {
+		if (is_type_compatible(p_right_type, p_left_type)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	return false;
```

Comment:
Should be possible to simplify as follows, right?

```suggestion
	if (!is_type_compatible(p_left_type, p_right_type, true) && is_type_compatible(p_right_type, p_left_type)) {
		return true;
	}
	return false;
```

Reply:
Also, do you need the first `if`, or would it be sufficient to check `is_type_compatible(p_right_type, p_left_type)`?

Reply:
@anvilfolk I removed the `is_super_type` method now completely.